pkg/resources/stream_common.go

```diff
@@ -134,3 +141,74 @@ var beforeSchema = &schema.Schema{
 	},
 	ConflictsWith: []string{"at"},
 }
+
+func handleStreamTimeTravel(d *schema.ResourceData) *sdk.OnStreamRequest {
+	if v := d.Get(AtAttributeName).([]any); len(v) > 0 {
+		return sdk.NewOnStreamRequest().WithAt(true).WithStatement(handleStreamTimeTravelStatement(v[0].(map[string]any)))
+	}
+	if v := d.Get(BeforeAttributeName).([]any); len(v) > 0 {
+		return sdk.NewOnStreamRequest().WithBefore(true).WithStatement(handleStreamTimeTravelStatement(v[0].(map[string]any)))
+	}
+	return nil
+}
+
+func handleStreamTimeTravelStatement(timeTravelConfig map[string]any) sdk.OnStreamStatementRequest {
+	statement := sdk.OnStreamStatementRequest{}
+	if v := timeTravelConfig["timestamp"].(string); len(v) > 0 {
+		statement.WithTimestamp(v)
+	}
+	if v := timeTravelConfig["offset"].(string); len(v) > 0 {
+		statement.WithOffset(v)
+	}
+	if v := timeTravelConfig["statement"].(string); len(v) > 0 {
+		statement.WithStatement(v)
+	}
+	if v := timeTravelConfig["stream"].(string); len(v) > 0 {
+		statement.WithStream(v)
+	}
+	return statement
+}
+
+func DeleteStreamContext(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+	id, err := sdk.ParseSchemaObjectIdentifier(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	err = client.Streams.Drop(ctx, sdk.NewDropStreamRequest(id).WithIfExists(true))
+	if err != nil {
+		return diag.Diagnostics{
+			diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "Error deleting stream",
+				Detail:   fmt.Sprintf("id %v err = %v", id.Name(), err),
+			},
+		}
+	}
+
+	d.SetId("")
+	return nil
+}
+
+func handleStreamRead(d *schema.ResourceData,
+	id sdk.SchemaObjectIdentifier,
+	stream *sdk.Stream,
+	streamDescription *sdk.Stream,
+) error {
+	if err := d.Set("comment", stream.Comment); err != nil {
```

Comment:
nit: we could use errors.Join(...) and return just one error here. Maybe it does not help that much with only 4 params but at least we return all the errors instead of just the first encountered one

Reply:
I'll do this in the next PR.